libs/watt/src/lib/components/drawer/watt-drawer.component.ts

```diff
@@ -57,28 +57,48 @@ export type WattDrawerSize = 'small' | 'normal' | 'large';
   imports: [A11yModule, MatSidenavModule, WattButtonComponent, WattSpinnerComponent],
 })
 export class WattDrawerComponent implements OnDestroy {
-  private cdr = inject(ChangeDetectorRef);
   private elementRef = inject(ElementRef);
   private overlayContainer = inject(OverlayContainer);
-  private cssCustomPropertiesService = inject(WattCssCustomPropertiesService);
+  private cdkTrapFocus = viewChild.required(CdkTrapFocus);
+  private bypassClickCheck = false;
+  private writableIsOpen = signal(false);
+
+  // Multiple drawers open at the same time is not allowed. This keeps track of
+  // the currently opened drawer and closes it when a new drawer is opened.
   private static currentDrawer?: WattDrawerComponent;
 
   /** Used to adjust drawer size to best fit the content. */
   size = input<WattDrawerSize>('normal');
 
-  /** Whether the drawer is open.  */
-  isOpen = model(false);
+  /**
+   * Whether the drawer should open automatically. If `key` is provided and
+   * `autoOpen` is true, the drawer will open every time the key changes.
+   */
+  autoOpen = input(false, { transform: booleanAttribute });
```

Comment:
question: the `input` is boolean. Is it still necessary to have the `booleanAttribute` then?

Reply:
Not strictly necessary, no, but I like that you can omit the boolean (`<watt-drawer autoOpen />` instead of `<watt-drawer [autoOpen]="true" />`). It also coerces any value to boolean, so you can do fancy stuff (perhaps too fancy 😄) like this:

```html
<watt-drawer [autoOpen]="id()" />
```

Resulting in a drawer that only "auto opens" if the id is truthy